mentoring/mentoring.py

```diff
@@ -144,14 +177,31 @@ def student_view(self, context):
         # Migrate stored data if necessary
         self.migrate_fields()
 
-        fragment, named_children = self.get_children_fragment(
-            context, view_name='mentoring_view',
-            not_instance_of=self.FLOATING_BLOCKS,
-        )
+        title = ""
+        header = ""
+
+        fragment = Fragment()
+        child_content = u""
+        for child_id in self.children:
+            child = self.runtime.get_block(child_id)
+            if isinstance(child, TitleBlock):
```

Comment:
@e-kolpakov I've simplified the code a bit, but I believe this approach is more efficient than something like self._split_children() because this way only one child block is ever loaded at a time. If we split the sorting logic out into another method, then it will have to load all the children at once (unless I make the code significantly more complicated).